webapp/sources/ldap-inventory/inventory-repository/src/test/scala/com/normation/inventory/ldap/core/TestNodeUnserialisation.scala

```diff
@@ -195,5 +275,114 @@ class TestNodeUnserialisation extends Specification {
         )
       ))
     }
+
+    "correctly unserialize custom properties from 7_0" in {
+      import Json.*
+      node(linux70Ldif).customProperties must containTheSameElementsAs(
+        List(
+          CustomProperty("simpleString", Str("a simple string")),
+          CustomProperty(
+            "someJson",
+            Obj(
+              Chunk(
+                ("foo" -> Str("bar")),
+                ("i"   -> Num(42)),
+                ("b"   -> Bool(true)),
+                ("arr" -> Arr(Num(1), Num(2)))
+              )
+            )
+          )
+        )
+      )
+    }
+
+    "correctly unserialize environment variable" in {
+      node(linux70Ldif).environmentVariables must containTheSameElementsAs(
+        List(
+          EnvironmentVariable("UPSTART_INSTANCE"),
+          EnvironmentVariable("SHELL", Some("/bin/bash"))
+        )
+      )
+    }
+
+    "correctly unserialize processes" in {
+      node(linux70Ldif).processes must containTheSameElementsAs(
+        List(
+          Process(
+            193,
+            Some("/var/rudder/cfengine-community/bin/cf-serverd"),
+            Some(0.0f),
+            Some(0.0f),
+            Some("2014-09-25 09:45"),
+            Some("?"),
+            Some("root"),
+            Some(36664)
+          )
+        )
+      )
+    }
+
+    "7.0 and 8.3 ldif lead to the same node" in {
+      val n1 = node(linux70Ldif)
+      val n2 = node(linux83Ldif)
+      n1 === n2
+    }
+
+    "idempotence of unser/ser for 8.3" in {
+      val nodeEntry = new LDAPEntry(new Entry(linux83Ldif.split("\n").toSeq*))
+      val prog      = for {
+        n <- mapper.nodeFromEntry(nodeEntry)
+        e  = mapper.treeFromNode(n).root
+        p  = mapper.processesFromNode(n)
+        _  = e.resetValuesTo(LDAPConstants.A_PROCESS, p*)
+      } yield e
+
+      val entry = ZioRuntime.unsafeRun(prog)
+      val diff  = com.unboundid.ldap.sdk.Entry.diff(nodeEntry.backed, entry.backed, false, true)
+      diff.toArray() must beEmpty
+    }
+
+    "correctly unserialize a node even with malformed env var, process, soft update" in {
```

Comment:
okay, I guess if someone is having this problem we haven't heard about it yet :innocent: